File: Ruby/Stack.rb

class Stack
    def initialize()
        @stack = []
    end
    
    def push(value)
        @stack.push(value)
    end
    
    def pop()
        @stack.pop
    end
    
    def display()
        print @stack.to_s + "\n"
    end
end

s = Stack.new()

s.push(1)
s.push(2)
s.push(3)
s.push(4)
s.display()

s.pop()
s.pop()
s.display()

File: src/swift/stack.swift
//
//  stack.swift
//
//
//  Created by Matheus Torres on 14/11/20.
//

class Stack {
    var stack: [Int] = []

    func push(_ value: Int) {
        stack.append(value)
    }

    func pop() {
        stack.removeLast()
    }

    func show() {
        print(stack)
    }

    func clean() {
        stack = []
    }
}

let stack = Stack()

stack.push(1)
stack.push(2)
stack.push(3)
stack.push(4)

stack.show()

stack.pop()
stack.pop()
stack.show()

